### ISSUE TITLE
[Suggestion] Allow event managers to add songs for the round to the jukebox

### ISSUE THREAD

**Issue by SatinIsle** [VOREStation/VOREStation#15993]
I know this isn't how issues are generally used, but Kashargul suggested I open one so they don't forget about it, so a little exception here because this would be real handy for the future:

Requires: admin
-> button to create a new song datum
-> category for admins only to play them

**Comment by ItsSelis** [comment]
What I can add to this, is that the "Media Tracks" subsystem manages the tracks that will be made available to jukeboxes. It will have a list called jukebox_tracks.

Iirc, it is possible to append stuff to that list (maybe a function inside of the subsystem can be added). However, so far I had to vv every jukebox to call a specific loading proc (I don't recall which one exactly it was) to reload all the tracks from the subsystem.

But in theory it is possible to use these steps to add custom jukebox tracks.

Also: Something similar is possible for only specific jukeboxes, as they probably also have a list of tracks saved in their own variables. Those can be probably altered as well, if wanted (so far this only is a feature for secret jukeboxes or something I think).

**Comment by Kashargul** [comment]
> What I can add to this, is that the "Media Tracks" subsystem manages the tracks that will be made available to jukeboxes. It will have a list called jukebox_tracks.
> 
> Iirc, it is possible to append stuff to that list (maybe a function inside of the subsystem can be added). However, so far I had to vv every jukebox to call a specific loading proc (I don't recall which one exactly it was) to reload all the tracks from the subsystem.
> 
> But in theory it is possible to use these steps to add custom jukebox tracks.
> 
> Also: Something similar is possible for only specific jukeboxes, as they probably also have a list of tracks saved in their own variables. Those can be probably altered as well, if wanted (so far this only is a feature for secret jukeboxes or something I think).
> 
> tl;dr:
> 
> 1. Update subsystem tracks list
> 2. Call jukebox update proc
> 3. **(OPTIONAL)** Repair jukebox flags (if it has 0 tracks it will be flagged as broken). This most likely will not be the case, but still good to know/implement!

The subsystem already has a manual track add with tgui inputs. So it's just to make it available to the interface. Maybe adjust the function to directly take inputs from the UI to save the tgui inputs or make a second one with parameters. So not too much work luckily. Should have some time to look into it this weekend.
